Duplicati/Library/Backend/SSHv2/SSHv2Backend.cs

```diff
@@ -336,42 +341,44 @@ public IEnumerable<IFileEntry> List()
+            foreach (var ls in m_con.ListDirectory(path))
+            {
+                if (ls.Name.ToString() == "." || ls.Name.ToString() == "..") continue;
+                yield return new FileEntry(ls.Name, ls.Length,
+                    ls.LastAccessTime, ls.LastWriteTime) { IsFolder = ls.Attributes.IsDirectory };
+            }
         }
 
-        public static Renci.SshNet.PrivateKeyFile ValidateKeyFile(string filename, string password)
+        private static Renci.SshNet.PrivateKeyFile ValidateKeyFile(string filename, string password)
         {
             try
             {
-                if (filename.StartsWith(KEYFILE_URI, StringComparison.OrdinalIgnoreCase))
+                if (!filename.StartsWith(KEYFILE_URI, StringComparison.OrdinalIgnoreCase))
                 {
-                    using (var ms = new System.IO.MemoryStream())
-                    using (var sr = new System.IO.StreamWriter(ms))
-                    {
-                        sr.Write(Duplicati.Library.Utility.Uri.UrlDecode(filename.Substring(KEYFILE_URI.Length)));
-                        sr.Flush();
-
-                        ms.Position = 0;
-
-                        if (String.IsNullOrEmpty(password))
-                            return new Renci.SshNet.PrivateKeyFile(ms);
-                        else
-                            return new Renci.SshNet.PrivateKeyFile(ms, password);
-                    }
+                    return String.IsNullOrEmpty(password)
+                        ? new PrivateKeyFile(filename)
+                        : new PrivateKeyFile(filename, password);
                 }
-                else
+
+                using (var ms = new System.IO.MemoryStream())
+                using (var sr = new System.IO.StreamWriter(ms))
                 {
-                    if (String.IsNullOrEmpty(password))
-                        return new Renci.SshNet.PrivateKeyFile(filename);
-                    else
-                        return new Renci.SshNet.PrivateKeyFile(filename, password);
+                    sr.Write(Utility.Uri.UrlDecode(filename.Substring(KEYFILE_URI.Length)));
+                    sr.Flush();
+
+                    ms.Position = 0;
+
+                    return String.IsNullOrEmpty(password) ? 
+                        new PrivateKeyFile(ms) : new PrivateKeyFile(ms, password);
                 }
+
             }
             catch (Exception ex)
             {
-                throw new UserInformationException(string.Format("Failed to parse the keyfile, check the key format and passphrase. Error message was {0}", ex.Message), "SSHFailedToParseKeyFile", ex);
+                throw new UserInformationException(
+            "Failed to parse the keyfile, check the key format and passphrase. " +
+                    $"Error message was {ex.Message}",
+                    "SSHFailedToParseKeyFile", ex);

```

Comment:
What IDE are you using?  This seems like an interesting formatting choice...

Reply:
Rider, though the string interpolation was me.

Reply:
Is the indentation a bit strange on the first line?